CIPs/cip-eth-compat.md

```diff
@@ -0,0 +1,95 @@
+- Many users coming from Ethereum are already familiar with wallets such as Metamask, and would likely prefer transacting on Celo using these wallets rather than having to install Celo-specific forks or completely new software such as `celocli`.
+- Users of cross-chain bridges, in particular, will need to interact with both the Celo blockchain and other blockchains (usually Ethereum).  The ability to use a single wallet to interact with both Celo and Ethereum would reduce the friction in doing this.
+- Compatibility with Ethereum signers such as Metamask would enable dapp developers to make use of [Web3Modal](https://github.com/Web3Modal/web3modal), easing the process of porting Ethereum dapps to Celo.
+
+## Specification
+
+This change requires activation as part of a hard fork.  As of `FORK_BLOCK_NUMBER` clients MUST accept transactions in either of two formats:
+
+1. Normal Celo transactions
+2. Ethereum-compatible transactions
+
+Note that the Normal Celo transactions format simply represents Celo transactions as they currently exist.  The only change to them is an additional field when represented as JSON (see below).
+
+### RLP transaction format
+
+When RLP-encoded, the two transaction formats consist of the following RLP lists:
+
+1. Normal Celo transactions: `(nonce, gasprice, gaslimit, feecurrency, gatewayfeerecipient, gatewayfee, recipient, amount, data, v, r, s)`
+2. Ethereum-compatible transactions: `(nonce, gasprice, gaslimit, recipient, amount, data, v, r, s)`
+
+### Transaction signing and signature verification
+
+For the purpose of transaction signing and of validating transaction signatures, clients MUST use the following as the "signing data" (which is RLP-encoded and hashed, and the result of that is then signed):
+
+1. For normal Celo transactions: `(nonce, gasprice, gaslimit, feecurrency, gatewayfeerecipient, gatewayfee, recipient, amount, data, chainid, 0, 0)`
+2. For Ethereum-compatible transactions: `(nonce, gasprice, gaslimit, recipient, amount, data, chainid, 0, 0)`
+
+### Transaction validation and processing
+
+When validating and processing transactions, clients MUST treat Ethereum-compatible transactions as equivalent to normal Celo transactions with the following values filled in for the missing fields:
+
+```
+feecurrency: nil
+gatewayfeerecipient: nil
+gatewayfee: 0
+```
+
+### JSON representation
+
+When representing or parsing transactions in JSON format, clients SHOULD support an additional boolean field called `ethCompatible`, which indicates whether the transaction is an Ethereum-compatible transaction (`true`) or a normal Celo transaction (`false`).  This is necessary in order to know what should be used as the transaction's "signing data" (as described above).  If the field is absent in the JSON being parsed, clients SHOULD treat the transaction as if `ethCompatible` were `false`.
```

Comment:
what should node rpc behavior in `signTransaction` and `sendTransaction` be when `ethCompatible` is set to `true` but one of the 3 celo-specific fields is specified as a non-default value

Reply:
can we specify which RPC methods accept or return JSON transactions?

Reply:
Good points!  I will:

(a) add `ethCompatible` as an option for `eth_sendTransaction` and `eth_signTransaction`
(b) add validation in these methods, in the tx pool, and in the state transition to reject transactions with `ethCompatible` set to `true` if they have non-(nil or 0) values for the 3 Celo-only fields
(c) Update the CIP with these details
(d) Look into and call out these and other methods rather accept or receive JSON transactions (or 'partial' transaction JSON objects such as the arguments to `eth_signTransaction`)

Reply:
Added a commit making this section more complete and (I hope) clearer.  I didn't list all the RPC methods that include transactions in their results, but gave a couple as examples.

I also pushed to the celo-blockchain branch some commits to add (a) handling for `ethCompatible` in RPC methods which take in a transaction in JSON format, (b) validation against `ethCompatible` being true when the transaction has non-nil-or-0 values for the Celo-only fields.